libs/foundry-dev-tools/src/foundry_dev_tools/config/token_provider.py

```diff
@@ -119,7 +119,7 @@ def __init__(
         client_id: str,
         client_secret: str | None = None,
         grant_type: FoundryOAuthGrantType | None = None,
-        scopes: list[str] | None = None,
+        scopes: list[str] | str | None = None,
```

Comment:
Yes, I was just testing different options to overwrite scopes and I think I added this so that environment variable passed scopes (comma separated string) get picked up correctly.
I also added a different method to parse true, false as booleans.